### PR TITLE
indexer: Location indexer and iloc re-index

### LINKED ISSUE #186: Re-index ilocs when editing loc
When editing any field of a location, we should re-index the internal locations that refer to it.
In general, we have to check all the `$ref` and re-index when the data is updated using the record editor or anything else.

### PR DESCRIPTION
**Pull request by topless**
* when updating a location property, we re-index all internal locations referring to it
* changed bulk_index to explicit index each record
* resolves #186

### CODE REVIEW

File: invenio_app_ils/indexer.py [hunk 1]
```diff
@@ -19,20 +19,19 @@
...
     """Index loan to refresh item reference."""
     loan_search = search_by_pid(item_pid=item_pid)
-    loan_ids = []
     for loan in loan_search.scan():
         record = Loan.get_record_by_pid(loan[Loan.pid_field])
         if record:
-            loan_ids.append(record.id)
-
-    RecordIndexer().bulk_index(loan_ids)
+            RecordIndexer().index(record)
```
Comment (zzacharo) [inline invenio_app_ils/indexer.py]: Why not indexing asynchronous here?

File: invenio_app_ils/indexer.py [hunk 2]
```diff
@@ -50,28 +49,25 @@ class ItemIndexer(RecordIndexer):
...
     """Index item to refresh document reference."""
     item_pids = circulation_items_retriever(document_pid)
-    item_ids = []
     for pid in item_pids:
         record = Item.get_record_by_pid(pid)
-        item_ids.append(record.id)
-
-    RecordIndexer().bulk_index(item_ids)
+        if record:
+            RecordIndexer().index(record)
```
Comment (zzacharo) [inline invenio_app_ils/indexer.py]: same here.

File: invenio_app_ils/indexer.py (hunk 1 above)
Comment (topless) [reply invenio_app_ils/indexer.py]: According to @ntarocco, we don't want to wait for the 5 minutes for the task to be picked up. But this is just our idea, feel free to elaborate.

File: invenio_app_ils/indexer.py (hunk 2 above)
Comment (topless) [reply invenio_app_ils/indexer.py]: Same here :p

File: invenio_app_ils/indexer.py (hunk 1 above)
Comment (zzacharo) [reply invenio_app_ils/indexer.py]: where is defined the 5 minutes? I assume you mean the indexing task?

Comment (ntarocco) [reply invenio_app_ils/indexer.py]: `bulk_index` is adding to the queue, then we need to `process` the queue or wait for 5 minutes.
See: https://github.com/inveniosoftware/invenio-app-ils/blob/master/invenio_app_ils/cli.py#L323

@zzacharo WDYT?
In this way, it is already async with `@shared_task`

Comment (zzacharo) [reply invenio_app_ils/indexer.py]: I mean is already asynchronous the task so this assumes that we don't really waiting for the result either way. But I see the point of not waiting, not sure if is a problem. I think that the ES is more performant in the bulk index form but I am not sure though. We can change it and see if it cause troubles or not in the future.

**Review by zzacharo — DISMISSED** [review]

File: invenio_app_ils/indexer.py (hunk 1 above)
Comment (ntarocco) [reply invenio_app_ils/indexer.py]: Agree, it just that the process_queue triggered an error when we tried (we did not investigate).
@topless do you want to try that? It is a little performance gain.

**Pull request merged**